MarsRover.Tests/ControllerShould.cs

```diff
@@ -25,48 +25,17 @@ public void Setup_ReturnsRoverLocationAndDirection_GivenValidMapAndRoverData()
                 .Returns("2,0");
             var output = new StubOutput();
             var mockMapInput = new Mock<IMapInput>();
-            mockMapInput.Setup(_ => _.Read("TestFiles/validfile1.txt"));
-            var map = SetMap();
-            var rover = new Rover(Direction.North, 2, 0);
+            mockMapInput.Setup(_ => _.Read("TestFiles/validFile1.txt"))
+                .Returns("ONNN\nNNNN\nNNNN");
+            mockMapInput.Setup(_ => _.FileExists("TestFiles/validFile1.txt"))
+                .Returns(true);
+            var controller = new Controller(mockInput.Object, output, mockMapInput.Object); 
 
-            var controller = new Controller(mockInput.Object, output, mockMapInput.Object, map, rover); 
+            controller.Setup(); 
 
-            controller.Setup(); // TODO: added this line & now test hangs. filepath is not valid (used debugger)
-
-            rover.Direction.Should().Be(Direction.North);
-            rover.Location.X.Should().Be(2);
-            rover.Location.Y.Should().Be(0);
-        }
-
-        public static Map SetMap()
-        {
-            var width = 5;
-            var height = 4;
-            var squares = new List<Square>
-            {
-                new Square(SquareContent.Obstacle, 0, 0),
-                new Square(SquareContent.None, 1, 0),
-                new Square(SquareContent.None, 2, 0),
-                new Square(SquareContent.None, 3, 0),
-                new Square(SquareContent.None, 4, 0),
-                new Square(SquareContent.None, 0, 1),
-                new Square(SquareContent.None, 1, 1),
-                new Square(SquareContent.None, 2, 1),
-                new Square(SquareContent.None, 3, 1),
-                new Square(SquareContent.Obstacle, 4, 1),
-                new Square(SquareContent.None, 0, 2),
-                new Square(SquareContent.None, 1, 2),
-                new Square(SquareContent.Obstacle, 2, 2),
-                new Square(SquareContent.None, 3, 2),
-                new Square(SquareContent.None, 4, 2),
-                new Square(SquareContent.None, 0, 3),
-                new Square(SquareContent.None, 1, 3),
-                new Square(SquareContent.Obstacle, 2, 3),
-                new Square(SquareContent.None, 3, 3),
-                new Square(SquareContent.None, 4, 3)
-            };
-            var map = new Map(width, height, squares);
-            return map;
+            controller.Rover.Direction.Should().Be(Direction.North);

```

Comment:
is there much value in testing these things in the controller? The parsing is already covered in your parser tests and i remember seeing something similar already tested in your rover test class. There's some 'reaching in' going on here.